custom-pages/login/src/services/AuthApi.js

```diff
@@ -29,22 +29,22 @@ export default class AuthApi {
     this.instance = new auth0.WebAuth(params)
   }
 
-  async login(username, password) {
-    return this.instance.login(
+  login(username, password, cb) {
+    this.instance.login(
       {
         realm: authOpts.connection,
         username,
         password
       },
       (err) => {
         if (err) {
-          if (!err.message) {
-            err.message = 'Invalid username or password'
+          if (cb) {
+            setTimeout(() => cb('Invalid username or password'), 5)
```

Comment:
why the setTimeout?

Reply:
The callback is executed in a service outside of React but it needs triggering a React re-render (in the caller component) so I am forcing the (React) reconciler to go through another cycle using setTimeout